### PR TITLE
refactor(fdr2csv): reimplement fdr2csv in rust

### PR DESCRIPTION
**Pull request by lukecologne**
## Summary of Changes

This PR reimplements the FDR2CSV tool in rust. This is done to ease changes in the FDR file format, by utilizing the serde and csv libaries to automize the serialization of the structs, eliminating the need to manually create large header files and CSV writing functions.

To decompress the FDR files (if it is compressed), the new converter uses the Gzip implementation from the flate2 library.
Casting of the raw bytes to the type is performed using the Bytemuck library.
Bindings to the C++ structs are generated using bindgen.

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

File: tools/fdr2csv/src/csv_header_serializer.rs [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+}
+
+pub struct CsvHeaderSerializer {
+    // The output string will be populated after each elementary data type in the record.
+    output: String,
+
+    // The field name list will keep track of the "higher" level field names
+    field_name_list: Vec<String>,
+
+    delimiter: String,
```
Comment (Gurgel100) [inline tools/fdr2csv/src/csv_header_serializer.rs:34]:
```suggestion
    delimiter: char,
```

File: tools/fdr2csv/src/csv_header_serializer.rs [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+}
+
+impl CsvHeaderSerializer {
+    // This method will be called if an elementary data type has been encountered.
+    // The field name list will then be joined together with a period as separator,
+    // and at the end the delimiter will be appended.
+    fn serialize_scalar(&mut self) -> Result<()> {
+        self.output += &self.field_name_list.join(".");
+        self.output += &self.delimiter;
```
Comment (Gurgel100) [inline tools/fdr2csv/src/csv_header_serializer.rs:43]:
```suggestion
        self.output.push(self.delimiter);
```

File: tools/fdr2csv/src/error.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
+use std;
```
Comment (Gurgel100) [inline tools/fdr2csv/src/error.rs:1]:
```suggestion
```

File: tools/fdr2csv/src/main.rs
Comment (Gurgel100) [inline tools/fdr2csv/src/main.rs:1]: As there are too many change proposals to do it over the GitHub review feature I attached a diff file containing the proposed changes.
[main.rs.txt](https://github.com/user-attachments/files/16240149/main.rs.txt)

**Review by Gurgel100 — COMMENTED** [review]
Looks really good. I only made some proposals to improve it even more.

**Review by 2hwk — APPROVED** [review]

**Pull request merged**